### PR TITLE
feat: seperate mapproxy deployments

### PR DESCRIPTION
**Pull request by shimoncohen**

### CODE REVIEW

File: helm/values.yaml [hunk 1]
```diff
@@ -110,7 +110,30 @@ env:
 
 mapproxy:
   enabled: true
-  replicaCount: 1
+  wms:
```
Comment (CL-SHLOMIKONCHA) [inline helm/values.yaml:113]: wms configs should be added to mapproxy umbrella under 'raster-helm' - 'serving' - 'mapproxyDeployment' values, so we can control those configs easily

File: helm/values.yaml [hunk 2]
```diff
@@ -110,7 +110,30 @@ env:
...
+    resources:
+      enabled: true
+      value:
+        limits:
+          cpu: 200m
+          memory: 128Mi
+        requests:
+          cpu: 200m
+          memory: 128Mi
+  wmts:
```
Comment (CL-SHLOMIKONCHA) [inline helm/values.yaml:125]: wmts configs should be added to mapproxy umbrella under 'raster-helm' - 'serving' - 'mapproxyDeployment' values, so we can control those configs easily

File: helm/values.yaml [hunk 3]
```diff
@@ -167,15 +190,6 @@ mapproxy:
     tls: 
       enabled: true
       useExternal: ''
-  resources:
```
Comment (CL-SHLOMIKONCHA) [inline helm/values.yaml:170]: handle this change (remove it )  in our umbrella in 'raster-helm' - 'serving' - 'mapproxyDeployment'

**Review by CL-SHLOMIKONCHA — CHANGES_REQUESTED** [review]

File: helm/config/nginx.conf [hunk 4]
```diff
@@ -37,34 +41,37 @@ server {
...
+    }
+
+    location {{ .Values.mapproxy.route.path }}/service {
+        if ($arg_service = 'wmts') {
+            rewrite ^ {{ .Values.mapproxy.route.path }}/service/wmts last;
         }
-        if ($arg_request = "GetCapabilities") { # support blocking kvp format request of capabilities
-            set $no_cache "1";
+        if ($arg_service = 'wms') {
+            rewrite ^ {{ .Values.mapproxy.route.path }}/service/wms last;
```
Comment (CL-SHLOMIKONCHA) [inline helm/config/nginx.conf:61]: same Ingress issue

**Review by CL-SHLOMIKONCHA — CHANGES_REQUESTED** [review]

File: helm/config/nginx.conf [hunk 5]
```diff
@@ -37,34 +41,37 @@ server {
...
+        proxy_pass http://mapproxy-wmts;
+    }
+
+    location {{ .Values.mapproxy.route.path }}/wmts {
+        include nginx.mapproxy.conf;
+        include nginx.mapproxy.wmts.conf;
+        proxy_pass http://mapproxy-wmts;
+    }
+
+    location {{ .Values.mapproxy.route.path }} {
```
Comment (asafMasa) [inline helm/config/nginx.conf:73]: add description for each location purpose

File: helm/values.yaml [hunk 6]
```diff
@@ -110,7 +110,30 @@ env:
...
   enabled: true
-  replicaCount: 1
+  wms:
+    enabled: true
+    replicaCount: 1
+    resources:
+      enabled: true
+      value:
+        limits:
+          cpu: 200m
```
Comment (asafMasa) [inline helm/values.yaml:120]: wms resources should be as defined today after the sizing - as this resources will make them restart

**Review by asafMasa — CHANGES_REQUESTED** [review]

File: helm/config/nginx.conf [hunk 7]
```diff
@@ -37,34 +41,37 @@ server {
...
-        proxy_cache_background_update on; # Allows starting a background subrequest to update an expired cache item, while a stale cached response is returned to the client.
-        proxy_cache_use_stale error timeout updating http_500 http_502 http_503 http_504; # return tile from cache in case of server down 
-        proxy_cache_revalidate on; # Enables revalidation of expired cache items
-        proxy_cache_lock on; # minimize the number of accesses to proxied servers when populating a new cache element
-        proxy_cache_min_uses 3; # Sets the number of requests after which the response will be cached
-        set $no_cache "";
-        if ($request_uri ~* \.xml) { # support blocking restful format request of capabilities
-            set $no_cache "1";
+
+    location {{ .Values.mapproxy.route.path }}/service/wmts {
```
[the remark this answers is not in the document]
Comment (shimoncohen) [reply helm/config/nginx.conf]: Fixed.

File: helm/config/nginx.conf (hunk 4 above)
Comment (shimoncohen) [reply helm/config/nginx.conf]: Fixed.

File: helm/config/nginx.conf (hunk 5 above)
Comment (shimoncohen) [reply helm/config/nginx.conf]: Added.

File: helm/values.yaml (hunk 6 above)
Comment (shimoncohen) [reply helm/values.yaml:120]: These are values, they are made to change for every env.

File: helm/config/nginx.mapproxy.wmts.conf [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+proxy_cache_revalidate on; # Enables revalidation of expired cache items
+proxy_cache_lock on; # minimize the number of accesses to proxied servers when populating a new cache element
+proxy_cache_min_uses 3; # Sets the number of requests after which the response will be cached
+set $no_cache "";
+if ($request_uri ~* \.xml) { # support blocking restful format request of capabilities
+    set $no_cache "1";
+}
+if ($arg_request = "GetCapabilities") { # support blocking kvp format request of capabilities
+    set $no_cache "1";
+}
```
Comment (ronenkapelian) [inline helm/config/nginx.mapproxy.wmts.conf:17]: validate all cache situation:

1. on WMTS should cache tiles only
   a. after 3 times of "miss" the 4 time will be "hit". validate on logs + cache directory
   b. capabilities - bypass status for "cache_status" log for kvp and restfull - validate not store on cache directory
   2. cache not enabled on WMS

File: helm/config/nginx.mapproxy.conf [hunk 9]
```diff
@@ -0,0 +1,7 @@
+gzip on;
```
Comment (ronenkapelian) [inline helm/config/nginx.mapproxy.conf:1]: validate!
take 1-2 of the directive and check it working with chrome F12 mode (check the header it is gzip, etc)
i want to be sure the directives are enable and passing to the service

File: helm/templates/mapproxy-wms-deployment.yaml [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+{{- fail "When using external ssl secret the value for .Values.authentication.ssl.externalSecretName should be set" }}
+{{- end }}
+{{- end }}
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ $releaseName }}-{{ $type }}-deployment
+  annotations:
+    collectord.io/index: {{ quote .Values.splunkIndex }}
```
Comment (ronenkapelian) [inline helm/templates/mapproxy-wms-deployment.yaml:25]: there is exists lack (bug)
after the collectord.io/index annotation
you should add the annotation of:
"collectord.io/output: splunk"

also add to the wmts deployment the same

File: helm/values.yaml [hunk 11]
```diff
@@ -110,7 +110,30 @@ env:
...
+      value:
+        limits:
+          cpu: 100m
+          memory: 128Mi
+        requests:
+          cpu: 100m
+          memory: 128Mi
   image:
     repository: mapproxy
     tag: 'latest'
```
Comment (ronenkapelian) [inline helm/values.yaml:139]: on nginx.cache section on values.yaml (can't comment right there). consider rename or add indent to understand the cache only for WMTS deployment (mabye not should think about it)

File: helm/values.yaml (hunk 2 above)
Comment (shimoncohen) [reply helm/values.yaml:125]: Added.

File: helm/values.yaml (hunk 3 above)
Comment (shimoncohen) [reply helm/values.yaml:170]: Removed.

File: helm/config/nginx.mapproxy.wmts.conf [hunk 12]
```diff
@@ -0,0 +1,20 @@
+{{- if .Values.nginx.cache.enabled }}
+proxy_cache_key "$host$request_uri";
+proxy_cache {{ .Values.nginx.cache.keysZone }};
+proxy_cache_valid 200 {{ .Values.nginx.cache.expirationTime }}; # will cache only valid 200 status code responses
+proxy_cache_valid any 0; # will disable caching all not 200 status code responses
+proxy_cache_background_update on; # Allows starting a background subrequest to update an expired cache item, while a stale cached response is returned to the client.
+proxy_cache_use_stale error timeout updating http_500 http_502 http_503 http_504; # return tile from cache in case of server down 
+proxy_cache_revalidate on; # Enables revalidation of expired cache items
+proxy_cache_lock on; # minimize the number of accesses to proxied servers when populating a new cache element
+proxy_cache_min_uses 3; # Sets the number of requests after which the response will be cached
```
[the remark this answers is not in the document]
Comment (shimoncohen) [reply helm/config/nginx.mapproxy.wmts.conf:10]: Fixed.

File: helm/config/nginx.mapproxy.wmts.conf (hunk 8 above)
Comment (shimoncohen) [reply helm/config/nginx.mapproxy.wmts.conf:17]: Checked.

File: helm/templates/mapproxy-wms-deployment.yaml (hunk 10 above)
Comment (shimoncohen) [reply helm/templates/mapproxy-wms-deployment.yaml:25]: Added.

File: helm/templates/mapproxy-wmts-deployment.yaml [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+{{- fail "When using external ssl secret the value for .Values.authentication.ssl.externalSecretName should be set" }}
+{{- end }}
+{{- end }}
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ $releaseName }}-{{ $type }}-deployment
+  annotations:
+    collectord.io/index: {{ quote .Values.splunkIndex }}
```
[the remark this answers is not in the document]
Comment (shimoncohen) [reply helm/templates/mapproxy-wmts-deployment.yaml:25]: Added.

**Review by asafMasa — APPROVED** [review]

File: helm/config/nginx.mapproxy.wmts.conf (hunk 8 above)
Comment (ronenkapelian) [reply helm/config/nginx.mapproxy.wmts.conf:17]: 1.a was checked and you saw it working by counting the miss rate configured value? (3)

File: helm/values.yaml (hunk 11 above)
Comment (ronenkapelian) [reply helm/values.yaml:139]: ok

File: helm/config/nginx.mapproxy.wmts.conf (hunk 8 above)
Comment (shimoncohen) [reply helm/config/nginx.mapproxy.wmts.conf:17]: Yes 👍🏻

**Review by asafMasa — APPROVED** [review]

**Review by CL-SHLOMIKONCHA — APPROVED** [review]

**Review by AnguIar — APPROVED** [review]

**Pull request merged**